networks/suzuka/suzuka-client/src/load_soak_testing/mod.rs

```diff
@@ -0,0 +1,394 @@
+use std::{fs::File, sync::Arc};
+use tracing_subscriber::{filter, prelude::*};
+
+mod scenario;
+pub use scenario::Scenario;
+
+const EXEC_LOG_FILTER: &str = "exec";
+
+/// Initialize all test's components with the configuration.
+/// Must be call before the test start: execute_test
+pub fn init_test(config: &ExecutionConfig) -> Result<(), std::io::Error> {
+	//do some verification on the config
+	config.verify_config();
+
+	// init tracing to log error in a file and stdout
+	// and log execution data in a json file.
+	let stdout_log = tracing_subscriber::fmt::layer().pretty();
+
+	// A layer that logs error and warn event to a file.
+	let log_file = File::create(&config.logfile)?;
+	let file_log = tracing_subscriber::fmt::layer().with_writer(Arc::new(log_file));
+
+	// A layer that logs execution event to a file.
+	let exec_file = File::create(&config.execfile)?;
+	let execution_log = tracing_subscriber::fmt::layer().json().with_writer(Arc::new(exec_file));
+
+	tracing_subscriber::registry()
+		.with(
+			stdout_log
+				.with_filter(filter::LevelFilter::INFO)
+				.and_then(file_log.with_filter(filter::LevelFilter::WARN))
+				// Add a filter that rejects spans and
+				// events whose targets start with `exec`.
+				.with_filter(filter::filter_fn(|metadata| {
+					!metadata.target().starts_with(EXEC_LOG_FILTER)
+				})),
+		)
+		.with(
+			// Add a filter to the exec label that *only* enables
+			// events whose targets start with `exec`.
+			execution_log.with_filter(filter::filter_fn(|metadata| {
+				metadata.target().starts_with(EXEC_LOG_FILTER)
+			})),
+		)
+		.init();
+	tracing::info!("Load and Soak test inited with config:{config:?}");
+	Ok(())
+}
+
+/// Define how the test will be run:
+/// * kind: Type of test to run
+/// * logfile_path: the file where log WARN and ERROR are written
+/// * execfile_path: File where execution data are written to be processed later.
+/// * define the number of started scenario per client. nb_scenarios / nb_scenario_per_client define the number of client.
+#[derive(Clone, Debug)]
+pub struct ExecutionConfig {
+	pub kind: TestKind,
+	pub logfile: String,
+	pub execfile: String,
+	pub nb_scenario_per_client: usize,
+}
+
+impl ExecutionConfig {
+	fn verify_config(&self) {
+		match self.kind {
+			TestKind::Load { nb_scenarios } => {
+				assert!(
+					nb_scenarios >= self.nb_scenario_per_client,
+					"Number of running scenario less than the number if scenario per client."
+				);
+			}
+			TestKind::Soak { min_scenarios, max_scenarios, .. } => {
+				assert!(max_scenarios >= min_scenarios, "max scenarios less than min scenarios");
+				assert!(
+					min_scenarios >= self.nb_scenario_per_client,
+					"Number of min running scenario less than the number if scenario per client."
+				);
+			}
+		}
+	}
+}
+
+impl Default for ExecutionConfig {
+	fn default() -> Self {
+		let nb_scenarios: usize = std::env::var("LOADTEST_NB_SCENARIO")
+			.unwrap_or("10".to_string())
+			.parse()
+			.unwrap_or(10);
```

Comment:
Is there a way we can avoid calling unwrap_or twice in a row?

Reply:
No, I don't think there is. It would be logically unsafe.